### PR TITLE
add more unit tests for `DIETClassifiers` data creation functionality

### LINKED ISSUE #9876: 8705 / extend unit tests for `DIETClassifiers` data creation functionality
### What problem are you trying to solve?

improve tests for the data creation functionality of `DIETClassifier`, which includes training data preparation (which in itself includes the preparation of `label_data`) and the preparation of data for prediction

### What's your suggested solution?

_No response_

### Examples (if relevant)

_No response_

### Is anything blocking this from being implemented? (if relevant)

_No response_

### Definition of Done

- [ ] add utilities for generating pseudo random features that allow as much control over the featurization process  as needed to imitate our current `Featurizers`
- [ ] add re-usable function for generating  (featurized) dummy NLU data 
- [ ] add unit tests to `DIETClassifier` testing the training data creation, label data creation and data creation at inference time under different scenarious

### PR DESCRIPTION
**Pull request by ka-bu**
**Proposed changes**:
- closes #9876 9876

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: tests/nlu/dummy_data/dummy_features.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        return collected
+
+    @staticmethod
+    def compare_with_feature_arrays(
+        actual: Dict[Text, List[FeatureArray]],
+        expected: List[ConcatenatedFeaturizations],
+    ) -> None:
+        """Compares concatenated features."""
+
+        for feature_type in [FEATURE_TYPE_SENTENCE, FEATURE_TYPE_SENTENCE]:
```
Comment (usc-m) [inline tests/nlu/dummy_data/dummy_features.py:296]: I assume one of these is meant to be `SEQUENCE`?

**Review by usc-m — COMMENTED** [review]
LGTM mostly - spotted some stuff though

File: tests/nlu/dummy_data/dummy_features.py (hunk 1 above)
Comment (ka-bu) [reply tests/nlu/dummy_data/dummy_features.py:296]: great catch :)  really hard to distinguish, if we ever rename I'd vote for something like `STATIC` vs `SEQUENTIAL` or so (differs in lengths and letters :D)

**Review by usc-m — APPROVED** [review]
LGTM

**Pull request merged**